.github/workflows/ci.yml

```diff
@@ -4,31 +4,24 @@ on:
   - push
 
 jobs:
-  lint:
-    name: lint
+  push:
+    name: push
     runs-on: ubuntu-latest
 
     steps:
       - uses: actions/checkout@v2
-
       - uses: actions/setup-node@v2
         with:
           node-version: 16.x
           registry-url: https://registry.npmjs.org
 
-      - uses: actions/cache@v2
-        with:
-          path: ~/.npm
-          key: v1-${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
-          restore-keys: v1-${{ runner.os }}-node-
-
-      - run: npm install
-      - run: npm run build
-      - run: npm run lint:ci
+      - run: yarn install --frozen-lockfile
+      - run: yarn run build
+      - run: yarn run lint:ci
+      - run: yarn run test
 
-      # publish version tags
-      - name: npm publish
-        run: npm publish --access public
+      - name: Publish to NPM
         if: ${{ startsWith(github.ref, 'refs/tags/v') }}
+        run: yarn publish --access public
```

Comment:
do we break this out to a separate workflow or are we doing this like this now?

Reply:
unless the package has a complex CI it's simper to just run as a conditional step after all other steps succeed (since GH actions has no easy way to include a snippet)